policyeval/execute.go

```diff
@@ -28,6 +28,18 @@ func (pe *PolicyEvaluator) getRoomsUserIsIn(userID id.UserID) []id.RoomID {
 	return rooms
 }
 
+func (pe *PolicyEvaluator) updateServerCache(userID id.UserID, recommendations policylist.Recommendations) {
+	pe.policyServer.cacheLock.Lock()
+	eventCache := pe.policyServer.EventCache
+	pe.policyServer.cacheLock.Unlock()
+	for evtID, cache := range eventCache {
+		if cache.PDU.Sender == userID && cache.PDU.RoomID == recommendations.BanOrUnban.RoomID {
```

Comment:
This looks like it only updates the cache for the policy list room? Also, the method seems like it belongs on `PolicyServer` rather than `PolicyEvaluator`

Reply:
The lock is also needed for iterating, not just accessing the pointer to the map